src/backend/src/services/work-packages.services.ts

```diff
@@ -729,4 +730,68 @@ export default class WorkPackagesService {
 
     return workPackageTemplateTransformer(workPackage);
   }
+
+  static async editWorkPackageTemplate(
+    user: User,
+    workPackageTemplateId: string,
+    templateName: string,
+    templateNotes: string,
+    duration: number | undefined,
+    stage: WorkPackageStage | undefined,
+    blockedBy: BlockedByInfo[],
+    expectedActivities: string[],
+    deliverables: string[],
+    workPackageName: string | undefined
+  ): Promise<Work_Package_Template> {
+    if (!isAdmin(user.role)) throw new AccessDeniedGuestException('edit work package templates');
+
+    const originalWorkPackageTemplate = await prisma.work_Package_Template.findUnique({
+      where: {
+        workPackageTemplateId
+      },
+      ...workPackageTemplateQueryArgs
+    });
+
+    if (!originalWorkPackageTemplate) throw new NotFoundException('Work Package', workPackageTemplateId);
+    if (originalWorkPackageTemplate.dateDeleted) throw new DeletedException('Work Package', workPackageTemplateId);
+
+    const updatedBlockedBys = await (
```

Comment:
ok so we should update the blockedBys the same way we update reimbursementProducts. Look at updateReimbursementProducts method in reimbursement-request.utils.ts, the tl;dr is that you need to compare the new list of blockedBys to the old list of blockedBys. The blocked bys in the new list, not in the old list need to be created, the blocked bys in the old list but not in the new list, need to be deleted, and any blocked bys that are in both lists need to be updated with the new values

Reply:
I think you'll need to make a schema migration to add an optional dateDeleted to blockedByInfo